lib/AllocateDescriptorsPass.cpp

```diff
@@ -512,7 +512,11 @@ bool AllocateDescriptorsPass::AllocateKernelArgDescriptors(Module &M) {
       for (Argument &Arg : f_ptr->args()) {
         set_and_binding_list.emplace_back(kUnallocated, kUnallocated);
         if (discriminants_list[arg_index].index >= 0) {
-          set_and_binding_list.back().first = set;
+          if (clspv::GetArgKindForType(Arg.getType()) !=
+              clspv::ArgKind::PodPushConstant) {
+            // Don't assign a descriptor set to push constants.
+            set_and_binding_list.back().first = set;
+          }
           set_and_binding_list.back().second = binding++;
```

Comment:
Seems the binding assignment should be guarded too?

Reply:
The bindings are assigned in an unused descriptor set to make sure that unique discriminants are kept separate.